### DISCUSSION TITLE
2nd heater,3rd heater etc if there is a big gap between setpoint and real temprature.

### DISCUSSION THREAD

**Discussion by flippz** [jmcollin78/versatile_thermostat#265] [Ideas]
2nd heater,3rd heater etc if there is a big gap between setpoint and real temprature. So it will do 1st heater usually, but if the diff is high it will also use 2nd heater, if it is a even bigger gap then the 3rd.
I use the activity mode to switch between eco and comfort mode. But the gap is a bit big (17C to 22C) and with just one heater it takes to long. But when it just needs to "keep" the temperature one heater is enough. so if it could use the 2nd heater when the diff is big and not to keep i would not need to manually turn the 2nd heater on and off.

**Comment by jmcollin78** [comment]
Hello @flippz ,

> So it will do 1st heater usually, but if the diff is high it will also use 2nd heater, if it is a even bigger gap then the 3rd.

This is exactly what does the VTherm natively. If possible, it just shift the starts of the heater to avoid having all in // but after the starts , they will acts in //.

**Reply by flippz** [reply]
Ok I'll test it out...But what i understod was that they are not on at the same time but rather alternate. Will report back :)

**Reply by flippz** [reply]
![2023-12-11_17h56_16](https://github.com/jmcollin78/versatile_thermostat/assets/21279095/ddc14e2c-b4e9-4913-b2bf-efea4201377f)
As you can see from the attached picture it is not working as i hoped. It keeps alternating between the two heaters. I would like it to use 2nd heater ONLY when temp is far from the setpoint (2-3 degrees)

**Reply by jmcollin78** [reply]
It should definitively work. Here is how it is supposed to work:

![Capture d’écran 2023-12-11 à 18 08 37](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/641cc08f-32ba-41e5-b377-4f853a9ecdd1)


There is something weird in your configuration I guess:
1. the heater 2 starts at 17h25 only ?
2. are you sure they are all in the same VTherm ? Can you give me the attributes of the VTherm accessible in Development Tools / States (in yaml formatted with the <> button).

**Reply by flippz** [reply]
As far as i can see from  you picture it is not what i want. I would never want 2nd heater to be on if the temp is within +/- 1degree of setpoint.
I removed the 2nd heater now (its a noizy fan oven) but i will re add it and send the code to you

**Reply by flippz** [reply]
```
hvac_modes:
  - heat
  - "off"
min_temp: 10
max_temp: 28
target_temp_step: 0.1
preset_modes:
  - none
  - eco
  - comfort
  - boost
  - activity
friendly_name: Stueversatile
supported_features: 17
current_temperature: 21
temperature: 21
hvac_action: idle
preset_mode: activity
is_on: true
hvac_mode: heat
frost_temp: 0
eco_temp: 17
boost_temp: 28
comfort_temp: 21
frost_away_temp: 10
eco_away_temp: 17
boost_away_temp: 17
comfort_away_temp: 17
power_temp: 13
target_temperature_step: 0.1
ext_current_temperature: 4.37
ac_mode: false
saved_preset_mode: activity
saved_target_temp: 21
window_state: "off"
window_auto_state: "off"
window_bypass_state: false
security_delay_min: 15
security_min_on_percent: 0.7
security_default_on_percent: 0.25
last_temperature_datetime: "2023-12-11T18:15:07.932013+01:00"
last_ext_temperature_datetime: "2023-12-11T18:15:07.951310+01:00"
security_state: false
minimal_activation_delay_sec: 30
device_power: 1.5
mean_cycle_power: 0.255
total_energy: 178.49
last_update_datetime: "2023-12-11T18:15:19.731500+01:00"
timezone: Europe/Oslo
window_auto_enabled: false
motion_sensor_entity_id: binary_sensor.motion_sensor_motion_detection
temperature_unit: °C
is_device_active: false
ema_temp: 21
is_over_switch: true
is_inversed: false
underlying_switch_0: switch.ovn_switch_2
underlying_switch_1: switch.vifteovn_switch
on_percent: 0.17
power_percent: 17
on_time_sec: 49
off_time_sec: 250
cycle_min: 5
function: tpi
tpi_coef_int: 0.8
tpi_coef_ext: 0.01

```

**Comment by jmcollin78** [comment]
> As far as i can see from you picture it is not what i want. I would never want 2nd heater to be on if the temp is within +/- 1degree of setpoint.

OK. I don't understand this. This is too much specific to your case, and I will not do that.

Your configuratin seems good to me.

**Reply by flippz** [reply]
No worries. I just wanted it to not use that noizy heater if it is not nessacery. I'll just use it manually in stead

**Reply by jmcollin78** [reply]
There is some feature named "auxiliary heater" in the standard HA thermostat. Maybe you can have a look. I don't exactly what it does but it could be something near what you need.

**Comment by flippz** [comment]
Thank you 😊 I will take a look at it 👍👍

tir. 12. des. 2023, 20:17 skrev Jean-Marc Collin ***@***.***>:
